### DISCUSSION TITLE
Why passing $config to HttpLoggingFilter::shouldLog ?

### DISCUSSION THREAD

**Discussion by StefanNeuser** [bilfeldt/laravel-http-client-logger#28] [Q&A]
Hi, i'm wondering why you pass $config to shouldLog without using it like in HttpLogger

`$this->config = array_merge(config('http-client-logger'), $config);`

**Comment by bilfeldt** [comment]
@StefanNeuser I am not sure I follow what you mean.

In [`HttpLogger@log`](https://github.com/bilfeldt/laravel-http-client-logger/blob/48013049c7398b933c2acc2c16c839cc3e9e3711/src/HttpLogger.php#L29-L49) you can see that we use the config to determine the channel and disk.

**Reply by StefanNeuser** [reply]
Please compare this with your current version. You are passing $config but not using it. This is how it should looks like: 

```<?php

namespace Bilfeldt\LaravelHttpClientLogger;

use Illuminate\Support\Arr;
use Psr\Http\Message\RequestInterface;
use Psr\Http\Message\ResponseInterface;

class HttpLoggingFilter implements HttpLoggingFilterInterface
{
    public function shouldLog(
        RequestInterface $request,
        ResponseInterface $response,
        float $sec,
        array $context = [],
        array $config = []
    ): bool
    {

        $config = array_merge(config('http-client-logger'), $config);

        if (!Arr::get($config, 'enabled'))
        {
            return false;
        }

        if (Arr::get($config, 'filter_all'))
        {
            return true;
        }

        if (Arr::get($config, 'filter_2xx') && $response->getStatusCode() >= 200 && $response->getStatusCode() < 300)
        {
            return true;
        }

        if (Arr::get($config, 'filter_3xx') && $response->getStatusCode() >= 300 && $response->getStatusCode() < 400)
        {
            return true;
        }

        if (Arr::get($config, 'filter_4xx') && $response->getStatusCode() >= 400 && $response->getStatusCode() < 500)
        {
            return true;
        }

        if (Arr::get($config, 'filter_5xx') && $response->getStatusCode() >= 500 && $response->getStatusCode() < 600)
        {
            return true;
        }

        if (Arr::get($config, 'filter_slow') < $sec)
        {
            return true;
        }

        return false;
    }
}
```

**Reply by bilfeldt** [reply]
Ahh. So you are saying that `Bilfeldt\LaravelHttpClientLogger\HttpLogger` is correctly merging the packages config with the provided config, but `Bilfeldt\LaravelHttpClientLogger\HttpLoggingFilter` does not do that?

It seems you are correct. Would you be open to PR a fix for this (like your code above)?

**Reply by StefanNeuser** [reply]
You could just use that pcs of code :-)

**Reply by bilfeldt** [reply]
Feel free to pr 👍 This is not bug but a feature. And this is not a top prioritet for me at this point, so I will realistically not get this done any time soon.
